src/header.js

```diff
@@ -5,53 +5,35 @@ class Header extends Component {
   constructor(props) {
     super(props);
     this.state = {
-      searchActive: "",
+      searchActive: false,
       searchText: "",
     };
 
-    this.handleDiggClick = this.handleDiggClick.bind(this);
-    this.handleRedditClick = this.handleRedditClick.bind(this);
-    this.handleMashableClick = this.handleMashableClick.bind(this);
-    this.handleFeedrClick = this.handleFeedrClick.bind(this);
+    this.selectFeed = this.selectFeed.bind(this);
     this.handleSearchClick = this.handleSearchClick.bind(this);
     this.handleSearchChange = this.handleSearchChange.bind(this);
   }
 
-  handleSelectSource() {
-    this.setState({ selectSource:  "active" });
-
-  }
-
-  handleDiggClick() {
-    this.props.onSelect("Digg");
-  }
-
-  handleRedditClick() {
-    this.props.onSelect("Reddit");
-  }
-
-  handleMashableClick() {
-    this.props.onSelect("Mashable");
-  }
-
-  handleFeedrClick() {
-    this.props.onSelect("All");
+  selectFeed(feed) {
+    return () => {
+      this.props.onSelect(feed);
+    }
   }
 
   handleSearchClick() {
-    this.setState({ searchActive: this.state.searchActive ? null : "active" });
+    this.setState({ searchActive: !this.state.searchActive });
   }
 
   handleSearchChange(event) {
     this.setState({ searchText: event.target.value });
-    this.props.onSearch(this.state.searchText);
+    this.props.onSearch(event.target.value);
```

Comment:
There was a subtle bug here - because `setState` is asynchronous, when you called `this.props.onSearch`, it actually gets called with the previous state, since setState hasn't finished executing before hand.

You can also ensure that the function gets run after setState finishes by calling setState like this:

```
this.setState({ searchText: event.target.value }, () => {
   this.props.onSearch(this.state.searchText);
});
```